compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -446,270 +51,86 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
-    def shortValue: Short = a.shortValue
-    def intValue: Int = a.intValue
-    def doubleValue: Double = a.doubleValue
-    def charValue: Char = a.charValue
+    def _2: List[Tree] = field.elems
   }
 
-
-  implicit def treeHelper(a: Tree): TreeHelper = new TreeHelper {
-    def symbol: Symbol = a.symbol
-
-    def pos: Position = a.span
-
-    def isEmpty: Boolean = a.isEmpty
-
-    def tpe: Type = a.tpe
-
-    def exists(pred: (Tree) => Boolean): Boolean = a.find(pred).isDefined
+  abstract class DeconstructorCommon[T >: Null <: AnyRef] {
+    var field: T = null
+    def get: this.type = this
+    def isEmpty: Boolean = field eq null
+    def isDefined = !isEmpty
+    def unapply(s: T): this.type ={
+      field = s
+      this
+    }
   }
 
+}
 
-  implicit def annotHelper(a: Annotation): AnnotationHelper = new AnnotationHelper {
-    def atp: Type = a.tree.tpe
-
-    def assocs: List[(Name, Tree)] = assocsFromApply(a.tree)
-
-    def symbol: Symbol = a.tree.symbol
+object DottyBackendInterface {
 
-    def args: List[Tree] = List.empty // those arguments to scala-defined annotations. they are never emitted
+  private def erasureString(clazz: Class[_]): String = {
+    if (clazz.isArray) "Array[" + erasureString(clazz.getComponentType) + "]"
+    else clazz.getName
   }
 
-  def assocsFromApply(tree: Tree): List[(Name, Tree)] = {
-    tree match {
-      case Block(_, expr) => assocsFromApply(expr)
-      case Apply(fun, args) =>
-        fun.tpe.widen match {
-          case MethodType(names) =>
-            (names zip args).filter {
-              case (_, t: tpd.Ident) if (t.tpe.normalizedPrefix eq NoPrefix) => false
-              case _ => true
-            }
-        }
-    }
-  }
+  def requiredClass[T](implicit evidence: ClassTag[T], ctx: Context): Symbol =
+    ctx.requiredClass(erasureString(evidence.runtimeClass))
 
-  implicit def nameHelper(n: Name): NameHelper = new NameHelper {
-    def isTypeName: Boolean = n.isTypeName
-    def isTermName: Boolean = n.isTermName
-    def startsWith(s: String): Boolean = n.startsWith(s)
-    def mangledString: String = n.mangledString
+  def requiredModule[T](implicit evidence: ClassTag[T], ctx: Context): Symbol = {
+    val moduleName = erasureString(evidence.runtimeClass)
+    val className = if (moduleName.endsWith("$")) moduleName.dropRight(1)  else moduleName
+    ctx.requiredModule(className)
   }
 
-  implicit def symHelper(sym: Symbol): SymbolHelper = new SymbolHelper {
-    def exists: Boolean = sym.exists
+  extension symExtensions on (sym: Symbol) {
 
-    // names
-    def showFullName: String = sym.showFullName
-    def javaSimpleName: String = toDenot(sym).name.mangledString // addModuleSuffix(simpleName.dropLocal)
-    def javaBinaryName: String = javaClassName.replace('.', '/') // TODO: can we make this a string? addModuleSuffix(fullNameInternal('/'))
-    def javaClassName: String = toDenot(sym).fullName.mangledString // addModuleSuffix(fullNameInternal('.')).toString
```

Comment:
Having abstractions over names is very useful since it reduces the change we do the wrong thing and call `name.toString` for example, so I would keep methods like this around (no preference on where they should live and whether they should be extension methods)

Reply:
Done